bot/cogs/eggs/reaction.py

```diff
@@ -0,0 +1,27 @@
+from discord.ext.commands import Cog
+from datetime import datetime, timedelta
+
+
+class Reaction(Cog, name="Eggs"):
+    def __init__(self, client):
+        self.bot = client
+        self.THRESHOLD = timedelta(seconds=1.5)
+        self.mapping = {}
+
+    @Cog.listener("on_reaction_add")
+    async def catch(self, reaction, user):
+        self.mapping.setdefault(user, {})[reaction.message] = datetime.now()
```

Comment:
If I understand the Discord API documentation correctly, `reaction.message` here is a complex object, in which case I'd suggest using its global identifier as the key in the dictionary instead, because based on the code, you're never going to need the entire message itself ever again and that way you can spare a lot of memory.